### ISSUE TITLE
Deploy Git LFS ML Models to Production - Monday AM

### ISSUE THREAD

**Issue by micoverde** [micoverde/cultivate-uw-ml-mvp#218]
### Branch to Deploy
- **Branch**: `fix-00196-model-selection-settings`
- **Tag**: `v1.0.0-lfs-models`
- **Commit**: `d81cd32`

#### 🎯 Critical Fix
Fixes the CEQ misclassification issue where "Did the tower fall?" was incorrectly classified as OEQ with 99.68% confidence. Should be CEQ (closed-ended question).

#### 📦 New ML Models (Git LFS Tracked)
- **quick_ceq_fix_model.pth** (8.1KB) - 98% accuracy on critical CEQ tests ✅
- **ultra_oeq_ceq_model.pth** (13MB) - Transformer model with multi-head attention
- **oeq_ceq_improved_model.pth** (81KB) - Gradient descent with class weights
- Plus 10 additional models in `src/ml/trained_models/`

#### 🔧 Infrastructure Changes
1. **Git LFS Setup**: Now tracking .pth, .pkl, and other ML model files
2. **Dockerfile Updates**: Added git-lfs installation for proper model retrieval in builds
3. **.gitignore Updates**: Commented out .pkl and .pth to allow LFS tracking

### Deployment Steps for Monday AM

1. **Verify LFS Models Load Correctly**
   ```bash
   git checkout fix-00196-model-selection-settings
   git lfs pull
   ls -lah *.pth *.pkl
   ```

2. **Test Locally**
   ```bash
   # Test the quick CEQ fix model
   python -c "import torch; model = torch.load('quick_ceq_fix_model.pth'); print('Model loaded successfully')"
   
   # Test critical classification
   echo 'Did the tower fall?' | python test_classification.py
   # Should return: CEQ (not OEQ)
   ```

3. **Merge to Main**
   ```bash
   git checkout main
   git merge fix-00196-model-selection-settings
   git push origin main
   git push origin v1.0.0-lfs-models
   ```

4. **Deploy to Azure Container Apps**
   - The Dockerfile now includes git-lfs, so Azure builds will automatically pull LFS files
   - Trigger deployment with updated branch/tag

### Testing Checklist
- [ ] Verify "Did the tower fall?" → CEQ ✅
- [ ] Verify "What happened to the tower?" → OEQ ✅
- [ ] Check ensemble model uses new models
- [ ] Verify Azure build pulls LFS files correctly
- [ ] Monitor model performance metrics

### Important Notes
- **LFS Storage**: 13MB of model files now tracked via Git LFS
- **Build Time**: First build after merge may take longer due to LFS download
- **Rollback Plan**: Can revert to commit before `d81cd32` if issues arise

### References
- Issue #196: Model selection settings bug
- Issue #120: Fine-tuning specification
- Training scripts: `train_quick_ceq_fix.py`, `train_ultra_high_quality_model.py`

### Success Metrics
- CEQ classification accuracy > 95%
- "Did the tower fall?" correctly classified as CEQ
- No increase in API latency
- Ensemble model performance improved

---
**Priority**: HIGH 🔴
**Target Date**: Monday AM
**Owner**: @warrenjo
